File: HomeWork/Home10.1/src/main/Match.java
import java.util.Scanner;

public class Match {
    Team team1;
    Team team2;
    int goalsTeam1;
    int goalsTeam2;
    int yellowTeam1;
    int yellowTeam2;
    int redTeam1;
    int redTeam2;
    double topHeightTeam1;
    double topHeightTeam2;

    public Match(Team t1, Team t2) {
        this.team1 = t1;
        this.team2 = t2;
    }

    public void inputMatchData() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter goals scored by " + team1.name + ": ");
        goalsTeam1 = scanner.nextInt();
        System.out.println("Enter goals scored by " + team2.name + ": ");
        goalsTeam2 = scanner.nextInt();
        System.out.println("Enter yellow cards shown to " + team1.name + ": ");
        yellowTeam1 = scanner.nextInt();
        System.out.println("Enter yellow cards shown to " + team2.name + ": ");
        yellowTeam2 = scanner.nextInt();
        System.out.println("Enter red cards shown to " + team1.name + ": ");
        redTeam1 = scanner.nextInt();
        System.out.println("Enter red cards shown to " + team2.name + ": ");
        redTeam2 = scanner.nextInt();
        System.out.println("Enter top height reached by the ball for " + team1.name + ": ");
        topHeightTeam1 = scanner.nextDouble();
        System.out.println("Enter top height reached by the ball for " + team2.name + ": ");
        topHeightTeam2 = scanner.nextDouble();
    }

    public void updateTeams() {
        team1.updateStats(goalsTeam1, goalsTeam2, yellowTeam1, redTeam1, topHeightTeam1);
        team2.updateStats(goalsTeam2, goalsTeam1, yellowTeam2, redTeam2, topHeightTeam2);

        if (goalsTeam1 > goalsTeam2) {
            team1.points += 3.14;
            team2.points -= 0.5;
        } else if (goalsTeam1 < goalsTeam2) {
            team1.points -= 0.5;
            team2.points += 3.14;
        } else {
            team1.points += 2.71828;
            team2.points += 2.71828;
        }
    }
}

File: Class/practice10.1/Match.cpp
#include <iostream>
#include "Match.h"

Match::Match(Team &t1, Team &t2) : team1(t1), team2(t2) {}

void Match::inputMatchData() {
    std::cout << "Enter goals scored by " << team1.name << ": ";
    std::cin >> goals_team1;
    std::cout << "Enter goals scored by " << team2.name << ": ";
    std::cin >> goals_team2;
    std::cout << "Enter yellow cards shown to " << team1.name << ": ";
    std::cin >> yellow_team1;
    std::cout << "Enter yellow cards shown to " << team2.name << ": ";
    std::cin >> yellow_team2;
    std::cout << "Enter red cards shown to " << team1.name << ": ";
    std::cin >> red_team1;
    std::cout << "Enter red cards shown to " << team2.name << ": ";
    std::cin >> red_team2;
    std::cout << "Enter top height reached by the ball for " << team1.name << ": ";
    std::cin >> top_height_team1;
    std::cout << "Enter top height reached by the ball for " << team2.name << ": ";
    std::cin >> top_height_team2;
}

void Match::updateTeams() {
    team1.updateStats(goals_team1, goals_team2, yellow_team1, red_team1, top_height_team1);
    team2.updateStats(goals_team2, goals_team1, yellow_team2, red_team2, top_height_team2);

    if (goals_team1 > goals_team2) {
        team1.points += 3.14;
        team2.points -= 0.5;
    } else if (goals_team1 < goals_team2) {
        team1.points -= 0.5;
        team2.points += 3.14;
    } else {
        team1.points += 2.71828;
        team2.points += 2.71828;
    }
}
